File: app/src/main/java/com/zuk0/gaijinsmash/riderz/data/local/entity/bsa_response/Bsa.java
package com.zuk0.gaijinsmash.riderz.data.local.entity.bsa_response;


import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Element;

public class Bsa {

    @Attribute
    private int id;

    @Element(name="station", required = false)
    private String station;

    @Element(name="type", required = false)
    private String type;

    @Element(name="description", data = true)
    private String description;

    @Element(required = false)
    private String sms_text;

    @Element(required = false)
    private String posted;

    @Element(required = false)
    private String expires;

    public void setStation(String station) { this.station = station; }
    public void setDescription(String description) { this.description = description; }
    public void setType(String type) { this.type = type; }
    public String getType() { return type; }
    public String getStation() { return station; }
    public String getDescription() { return description; }

    public String getSms_text() {
        return sms_text;
    }

    public void setSms_text(String sms_text) {
        this.sms_text = sms_text;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getPosted() {
        return posted;
    }

    public void setPosted(String posted) {
        this.posted = posted;
    }

    public String getExpires() {
        return expires;
    }

    public void setExpires(String expires) {
        this.expires = expires;
    }
}

File: app/src/main/java/com/zuk0/gaijinsmash/riderz/data/local/entity/bsa_response/Bsa.kt
package com.zuk0.gaijinsmash.riderz.data.local.entity.bsa_response

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.zuk0.gaijinsmash.riderz.data.local.entity.station_response.Station
import org.simpleframework.xml.Attribute
import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

@Root(name = "bsa", strict = false)
data class Bsa
@JvmOverloads constructor(

    //@Attribute(required = false)
    @param:Element(name = "id", required = false)
    @field:Element(name = "id", required = false)
    var id: Int? = 0,

    @Expose
    @SerializedName("station")
   // @param:Element(name = "station", required = false)
    @field:Element(name = "station", required = false)
    var station: Station? = null,

    @Expose
    @SerializedName("type")
   // @param:Element(name = "type", required = false)
    @field:Element(name = "type", required = false)
    var type: String? = null,

    @Expose
    @SerializedName("description")
    //@param:Element(name = "description", data = true, required = false)
    @field:Element(name = "description", data = true, required = false)
    var description: String? = null,

    @Expose
    @SerializedName("sms_text")
    //@param:Element(name = "sms_text", required = false)
    @field:Element(name = "sms_text", required = false)
    var sms_text: String? = null,

    @Expose
    @SerializedName("posted")
    //@param:Element(name = "posted", required = false)
    @field:Element(name = "posted", required = false)
    var posted: String? = null,

    @Expose
    @SerializedName("expires")
    //@param:Element(name = "expires", required = false)
    @field:Element(name = "expires", required = false)
    var expires: String? = null

)